services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/internal/cookie/DefaultCookiesForCannedPolicy.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.cloudfront.internal.cookie;
+
+import java.net.URI;
+import java.util.Objects;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.services.cloudfront.cookie.CookiesForCannedPolicy;
+
+@Immutable
+@ThreadSafe
+@SdkInternalApi
+public final class DefaultCookiesForCannedPolicy implements CookiesForCannedPolicy {
+
+    private final String resourceUrl;
+    private final String keyPairIdHeaderValue;
+    private final String signatureHeaderValue;
+    private final String expiresHeaderValue;
+
+    private DefaultCookiesForCannedPolicy(DefaultBuilder builder) {
+        this.resourceUrl = builder.resourceUrl;
+        this.keyPairIdHeaderValue = builder.keyPairIdHeaderValue;
+        this.signatureHeaderValue = builder.signatureHeaderValue;
+        this.expiresHeaderValue = builder.expiresHeaderValue;
+    }
+
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new DefaultBuilder(this);
+    }
+
+    @Override
+    public String toString() {
+        return "CloudFront Cookies for Canned Policy:\n"
+               + "Resource URL = " + resourceUrl + "\n"
+               + "Signature Header Value = " + signatureHeaderValue + "\n"
+               + "Key-Pair-Id Header Value = " + keyPairIdHeaderValue + "\n"
+               + "Expires Header Value = " + expiresHeaderValue;
+    }
```

Comment:
Might be easier to use `ToString.builder()`?

Reply:
updated to use ToString